Client/Frontend/Autofill/CreditCard/ViewComponents/CreditCardInputField.swift

```diff
@@ -0,0 +1,209 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/
+
+import Foundation
+import SwiftUI
+import Combine
+import Shared
+
+enum CreditCardInputType {
+    case name, number, expiration
+}
+
+struct CreditCardInputField: View {
+    var fieldHeadline: String
+    var errorString: String
+    var delimiterCharacter: String?
+    var userInputLimit: Int
+    var formattedTextLimit: Int
+    var keyboardType: UIKeyboardType
+
+    @Binding var text: String
+    let inputType: CreditCardInputType
```

Comment:
same here, `let inputType: CreditCardInputType` would work best inside view model and then you can throughly test view model

Reply:
Basically all state and binding variables can live inside view model too, you aren't changing your view but only structuring it in a way where the view only has a single instance to deal with and not worry about variables